### DISCUSSION TITLE
SymbolDisplayFormat: use '+' to separate nested types

### DISCUSSION THREAD

**Discussion by weifenluo** [dotnet/roslyn#67798] [Q&A] [answered]
I'm writing a source generator which I need to store a string in `"{namespace}.{containingType}+{nestedType}+{...}"` format so that I can generate:

```
namespace {namespace}
{
    partial class {containingType}
    {
        partial class {nestedType}
        {
            partial class {...}
            {
            }
        }
    }
}
```
I need nested types to be separated by `'+'` instead of `'.'` so that I can get the first type by looking for last `'.'` in the string.

The string is generated from `INamedTypeSymbol`. `INamedTypeSymbol.ToDisplayString` which accepts a `SymbolDisplayFormat` parameter, seems the right way to go. There is a `SymbolDisplayCompilerInternalOptions.UsePlusForNestedTypes` exactly doing this, however, `SymbolDisplayFormat.CompilerInternalOptions` is internal.

Am I missing something?

**Comment by CyrusNajmabadi** [comment, marked as answer]
> I'm writing a source generator which I need to store a string in "{namespace}.{containingType}+{nestedType}+{...}" format so that I can generate:

You don't need to store a string for this. You could just store stuff like an list of tuples. Where each tuple contains the name and a boolean stating if it's a namespace or type to generate.
